grpcbp/server.go

```diff
@@ -0,0 +1,128 @@
+package grpcbp
+
+import (
+	"net"
+	"time"
+
+	"github.com/reddit/baseplate.go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/keepalive"
+)
+
+// ServerConfig is the argument struct for NewBaseplateServer. Please refer to
+// the documentation for each field to see how is it used.
+type ServerConfig struct {
+	// MaxConnectionIdle is a duration for the amount of time after which an idle
+	// connection would be closed by sending a GoAway. Idleness duration is
+	// defined since the most recent time the number of outstanding RPCs became
+	// zero or the connection establishment. The current default value is
+	// infinity.
+	MaxConnectionIdle time.Duration `yaml:"maxConnectionIdle"`
+
+	// MaxConnectionAge is a duration for the maximum amount of time a
+	// connection may exist before it will be closed by sending a GoAway. A
+	// random jitter of +/-10% will be added to MaxConnectionAge to spread out
+	// connection storms. The current default value is infinity.
+	MaxConnectionAge time.Duration `yaml:"maxConnectionAge"`
+
+	// MaxConnectionAgeGrace is an additive period after MaxConnectionAge after
+	// which the connection will be forcibly closed. The current defualt value is
+	// infinity.
+	MaxConnectionAgeGrace time.Duration `yaml:"maxConnectionAgeGrace"`
+
+	// After a duration of this time if the server doesn't see any activity it
+	// pings the client to see if the transport is still alive. If set below 1s,
+	// a minimum value of 1s will be used instead. The current default value is 2
+	// hours.
+	Time time.Duration `yaml:"time"`
```

Comment:
`Time` feels like a very confusing/non-descriptive name for it. Maybe `KeepaliveTime`?

Reply:
I agree the name is confusing.  KeepaliveTime is fine with me.  Also could go with a more descritive name like PingQuietClientsAfter.

Reply:
Maybe `PingInactiveClientsAfter`? The official names are all very misleading but I prefixed any keepalive related configuration with `KeepAlive` hoping that it makes it more clear.

Reply:
PingInactiveClientsAfter sounds fine, though I went with "Quiet" since it seems a bit clearer than "Inactive" (what counts as active?) but this is just bikeshedding.